CRM/Report/Form/Contribute/Summary.php

```diff
@@ -514,6 +532,15 @@ public function from($entity = NULL) {
           ON ({$this->_aliases['civicrm_batch']}.entity_id = eft.financial_trxn_id
           AND {$this->_aliases['civicrm_batch']}.entity_table = 'civicrm_financial_trxn')";
     }
+
+    // for credit card type
+    if ($this->isTableSelected('civicrm_financial_trxn')) {
+      $this->_from .= "\n LEFT JOIN civicrm_entity_financial_trxn eftcc
```

Comment:
should we be adding line breaks to the sql? I think I've seen it before but it feels wrong